pkg/operator/controller/controller.go

```diff
@@ -327,21 +321,21 @@ func (r *reconciler) ensureRouterNamespace() error {
 	return nil
 }
 
-// ensureClusterIngress ensures all necessary router resources exist for a
-// given clusteringress.
-func (r *reconciler) ensureClusterIngress(ci *ingressv1alpha1.ClusterIngress, caSecret *corev1.Secret, infraConfig *configv1.Infrastructure, dnsConfig *configv1.DNS, result *reconcile.Result) error {
+// ensureClusterIngress ensures all necessary router resources exist for a given clusteringress.
+func (r *reconciler) ensureClusterIngress(ci *ingressv1alpha1.ClusterIngress, caSecret *corev1.Secret, dnsConfig *configv1.DNS, infraConfig *configv1.Infrastructure) error {
 	errs := []error{}
 
 	ha := effectiveHighAvailability(ci, infraConfig)
 
 	deployment, err := r.ensureRouterDeployment(ci, infraConfig, ha)
 	if err != nil {
-		errs = append(errs, err)
+		errs = append(errs, fmt.Errorf("failed to ensure router deployment for %s: %v", ci.Name, err))
```

Comment:
Set `deployment = nil` here?

Reply:
You literally can't do anything right now without the deployment, so why not just keep returning early? And again, this is using the result of a fn call in the presence of error...